src/avl_tree/set.rs

```diff
@@ -0,0 +1,400 @@
+extern crate rand;
+
+use crate::avl_tree::tree::*;
+use rand::seq::SliceRandom;
+use rand::thread_rng;
+use std::mem::replace;
+use std::{cmp::Ordering, iter::FromIterator};
+
+#[derive(Debug, PartialEq, Clone)]
+struct AVLTreeSet<T: Ord> {
+    root: AVLTree<T>,
+}
+
+impl<'a, T: 'a + Ord> Default for AVLTreeSet<T> {
+    fn default() -> Self {
+        Self { root: None }
+    }
+}
```

Comment:
You could move the default method to the Tree itself and then add the `#[derive(Default)]` tag to the Set. But does it makes sense to have a Default method if it returns the same as `new`?

Reply:
Ah yes youre right, thanks. I don't know, but thats a best practice in many instantiable structures in rust

Reply:
Well then, doesn't hurt to have it.

Reply:
nah i remove it